pkg/netflow/portrollup/portrollup.go

```diff
@@ -0,0 +1,106 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package portrollup
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/netflow/common"
+	"sync"
+)
+
+// EndpointPairPortRollupStore contains port rollup states
+type EndpointPairPortRollupStore struct {
+	portRollupThreshold int
+	curStore            map[string]*portRollupTracker
+	newStore            map[string]*portRollupTracker
+
+	// mutex used to protect access to curStore and newStore
+	mu sync.Mutex
+}
+
+// NewEndpointPairPortRollupStore create a new *EndpointPairPortRollupStore
+func NewEndpointPairPortRollupStore(portRollupThreshold int) *EndpointPairPortRollupStore {
+	return &EndpointPairPortRollupStore{
+		// curStore and newStore map key is composed of `<SOURCE_IP>|<DESTINATION_IP>`
+		// SOURCE_IP and SOURCE_IP are converted from []byte to string.
+		// string is used as map key since we can't use []byte as map key.
+		curStore: make(map[string]*portRollupTracker),
+		newStore: make(map[string]*portRollupTracker),
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Can you add a comment about why we need a second store ? Took me some time to understand you need it because you reset it every `interval`, and you need to keep the past `interval` value to avoid resetting to 0. Also, this means that you are actually rolling up over a 2 * interval period, right ?

Reply:
Good idea. Added here 144607ee3